lib/internal/Magento/Framework/View/Element/Html/Link/Current.php

```diff
@@ -81,7 +85,7 @@ private function getMca()
      */
     public function isCurrent()
     {
-        return $this->getCurrent() || $this->getUrl($this->getPath()) == $this->getUrl($this->getMca());
```

Comment:
This change might be harmful since we stop using `getMca()` method at all. That means that we can have the current links functionality broken in different places. I would recommend adding an additional check here (by trimming `/index/index` or so) instead of removing invoking of `getMca()` method

Reply:
@rogyar thank you for suggestion, currently the logic of the `getMca` method was improved, by removing the `_defaultPath` logic (which was wrong in my opinion). Also there are covered all the cases when a default url part is used.

So having a new link `test/index`, the following possible cases will mark this link as **Current**:
- `test`
- `test/index`
- `test/index/index`

Reply:
Have you tested passing GET parameters via URL as well?

Reply:
Yes, with all these use cases and the link keeps its **current** state.